### ISSUE TITLE
fix: 🤔 clarify and/or fix interactive `sd-teaser`

### ISSUE THREAD

**Issue by karlbaumhauer** [solid-design-system/solid#1311]
## Current behavior
Caused by the need of several teams, we added a sample which shows the teaser where the teaser itself is a clickable element AND it contains another clickable element. 

=> https://solid-design-system.fe.union-investment.de/x.x.x/storybook/?path=/story/components-sd-teaser--samples-clickable

With this sample of the teaser we currently have 2 issues:
1. It is unclear if this is actually a solid solution we should provide
2. The story currently has 2 bugs:
   1. The tab-order does not work as expected. The first CTA is not being focused at all
   2. As we have 2 different hover zones inside the teaser, the teaser itself shows a hover effect (light gradient) and as soon as the user hovers over the CTA inside the teaser, it keeps the hover effect but adds the hover effect on the CTA to it => not clear to the user which element is being hovered.

## Expected behavior
1. We have clear understanding on what we want to achieve (and write it down)
2. If we keep the sample, the bugs need to be fixed

Regarding 2.2. => expected behavior needs to be defined by UX
## Steps to reproduce
1. open the URL https://solid-design-system.fe.union-investment.de/x.x.x/storybook/?path=/story/components-sd-teaser--samples-clickable 
2. click in the white space inside the story
3. start tabbing on the keyboard
## Technical Information
This topic has been part of a [PR](https://github.com/solid-design-system/solid/pull/670#discussion_r1447713677) and the respective [ticket](https://github.com/solid-design-system/solid/issues/503) in January but unfortunately, we did not write down any summary from our discussions we had.
## DoR
- [ ] Item has business value
- [ ] Item has been estimated by the team
- [ ] Item is clear and well-defined
- [ ] Item dependencies have been identified

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

**Comment by mariohamann** [comment]
I just created a proposal, how this could work out in future without violating specs and a11y:
https://codepen.io/mariohamann/full/wvLjdLr

@christophsaile helped me testing it and so far it looks good!

I would propose that we let test this in the accessibility audit instead of the other template.

**Comment by christophsaile** [comment]
@mariohamann a possible improvement could be that button/link also gets its focus state when the focus on the card is active, see last example here: https://codepen.io/aardrian/pen/vYOGpXb.
This way it is more clear that the card and the button / link are the same interaction

**Comment by matuzo** [comment]
This solution is definitely an improvement, but it's not perfect. The good news is there is no perfect solution. That pattern is still an unsolved problem. 

My two cents:
* I don't like the duplication of interactive elements the solution requires.
* While I don't have a clear picture of the final implementation, I urge caution. There could be a mismatch between the visible accessible name of the call to action component (link or button) and the hidden link or button. That could pose a problem for voice users. I know you purposely put "(hidden link)" there, but it illustrates the mismatch I'm talking about well. A voice user would say, "Click visit union investment," but potentially nothing would happen because the accessible name is "Visit Union Investment (Hidden Link)."
* Putting `tabindex=-1` and `aria-hidden=true` on interactive elements is generally not the best idea.
* This may or may not be an issue, but text isn't selectable.

You can address my first three concerns by using a pseudo-element instead. See this [CodePen for a demo](https://codepen.io/matuzo/pen/wvVMYMG).

```

sd-teaser{
  position: relative;
  margin: 8px;
  transform: translate(0); /* creates a new stacking context */
}

sd-button::part(base)::before {
  position: fixed;
  inset: 0;
  display: block;
  content: "";
}
```

If you want to make certain parts selectable or if they include other interactive elements, you can promote them to the top layer on the z-axis.

```
p {
  position: relative;
  z-index: 1;
}
```

**Comment by coraliefeil** [comment]
Why don’t we just go with solution 6 / pseudo element here?

https://codepen.io/matuzo/pen/ZEqbXOM

Seems to be the best option.

**Comment by coraliefeil** [comment]
> Why don’t we just go with solution 6 / pseudo element here?
> 
> https://codepen.io/matuzo/pen/ZEqbXOM
> 
> Seems to be the best option.
> 
> It would have to be adapted a little in the example so that both links are clickable.
> 
> The link would still need `position: relative` and a `z-index` , then it is also clickable

Just realized I already mentioned everything wrt clickable teaser here in this issue.

**Comment by mariohamann** [comment]
> Why don’t we just go with solution 6 / pseudo element here?
> 
> https://codepen.io/matuzo/pen/ZEqbXOM
> 
> Seems to be the best option.
> 
> It would have to be adapted a little in the example so that both links are clickable.
> 
> The link would still need `position: relative` and a `z-index` , then it is also clickable

That seems to work really well. I made an example – with the `:has()` selector so that you can even change the hover/focus style of the card depending on hover/focus of the single links in the component as well. I additionally enjoy, that hovering the whole card triggers the hover state of the button, so it makes the connect REALLY clear.

https://codepen.io/mariohamann/pen/yyLKMOM

<video src="https://github.com/user-attachments/assets/f2f00157-1dbb-4085-aa52-6bb258daecb5"></video>

**Comment by matuzo** [comment]
LGTM

**Comment by smfonseca** [comment]
Tackled on this [PR](https://github.com/solid-design-system/solid/pull/2120)
